R/aggregate.R

```diff
@@ -68,7 +73,7 @@ agg_climate_data <- function(year, data_raster, climate_var, daily_agg, trans =
   # Get layer names (dates)
   all_layers <- names(clim_raster)
   layer_names <- all_layers[seq(1, length(all_layers), 24)] # Keep every 24th layer name (1 per day)
-  layer_names <- paste0('month_', substring(layer_names, 7,8), '_day_', substring(layer_names, 10,11)) # Extract month/day for each layer
+  layer_names <- paste0('year_', substring(layer_names, 2,5), 'month_', substring(layer_names, 7,8), '_day_', substring(layer_names, 10,11)) # Extract month/day for each layer
```

Comment:
@tliddell4 is this going to work for any ERA5 variable? makes me nervous to have so much hard coded here.

Reply:
Yes I agree that's one of the things I plan to discuss today, both for ERA5 data and possibly an other data. There's a number of things we could do here, none of them too clear cut in my mind. We're either asking the user for a lot of information or just hoping the assumptions we make hold for the data they use.

Reply:
@tliddell4 I think we should make this functional for ERA5 data only. but is there a way to confirm nothing about these layers changes across ERA5 variables? and is there a way to code this a little more generally?

Reply:
I don't have much experience with ERA5 data, I think @saraorofino mentioned that these layer names are consistent though. We've had a few discussions about how to make this more flexible for the end package, but I can get this done for the beta version if this is a high priority.

Reply:
@tcarleton @tliddell4 I tried checking the ERA5 documentation to see if it mentioned the structures like this or not and I couldn't find anything. So far it's worked for the 3 variables we've tested but I don't know if that seems good enough to assume it's the same for all variables. Key considerations are: 

- Averaging over layers with the same date. Right now we use `raster::stackApply(clim_raster, indices = indices, fun=mean)` where `indices` is a vector of numbers the same length as the raster stack where the number associated with each layer is the group (e.g. take the mean of all the layers with the same index number). The indices argument has to be a number and it has to be the same length as the raster stack for the function to work but we could potentially create this vector based on the date instead of just assuming that the first 24 layers are one day, the next 24 a second day etc. However, I don't really know how to do this off the top of my head. 
- Extracting dates from layer names. If we do want to try and extract a date from layer names there is still one piece of hard coding that has to happen. From what I've seen all layer names are `Xyear.month.day.hours.minutes.seconds` (probably because you shouldn't or can't have layer names that are numbers?). We can use `lubridate::as_date()` to extract a date but we have to remove the "X" first. We can either explicitly remove X or we can remove the first character of the string hoping that it's an X or some other letter. Something like the following: 

```
all_layers <- names(clim_raster)
all_layers <- stringr::str_remove(all_layers, "X")
layer_dates <- lubridate::as_date(all_layers) # same length as raster stack - could be used to assigned indices?
layer_names <- unique(layer_dates) # 365 for a full year of raster data should be applied as column names later
```

- Dates as column headings. After the non-linear transformations we get a data.table that has 365 columns one for each layer (day). We apply this `layer_names` object as the column headings and then we pivot to a longer table collapsing those dates into a single date column. I quickly tried doing this and the dates don't get applied correctly to column headings (e.g. should be 20200101 but shows up as 18262) so that is something we would have to work out if we prefer to go this route. I think we might be able to just extract layer names as is do the transformations and then once the data.table is in long format remove the X and convert to a date but again we are still hard coding that one assumption that there is an X or another character ahead of the date.